draft-ietf-gnap-core-protocol.md

```diff
@@ -237,21 +237,24 @@ the information needed at a given stage is communicated out of band
 or is preconfigured between the components or entities performing
 the roles. For example, one entity can fulfil multiple roles, and so
 explicit communication between the roles is not necessary within the
-protocol flow.
-
-~~~
-        +------------+             +------------+
-        | End-user   | ~ ~ ~ ~ ~ ~ |  Resource  |
-        |            |             | Owner (RO) |
-        +------------+             +------------+
-            +                            +      
-            +                            +      
-           (A)                          (B)       
-            +                            +        
-            +                            +         
-        +--------+                       +       +------------+
-        | Client |--------------(1)------+------>|  Resource  |
-        |Instance|                       +       |   Server   |
+protocol flow. Additionally some components may not be involved
```

Comment:
A few more minor editorial suggestions...

The first is `Additionally, some components`. Then, the previous sentences already use `For example` a couple of times and you may want to replace it with `To illustrate` or something similar.

Lastly, I suggest using `a client instance can call the AS to get the direct user information and does not need to get an access token to call an RS.`.